src/firmware/app/control/python_controller/difference_equation.py

```diff
@@ -0,0 +1,109 @@
+#!/usr/bin/env python
+"""
+This file contains the implementation of the DifferenceEquation class which generates a difference equation based on an input discrete transfer function.
+
+"""
+
+import numpy as np
+import control as ct
+from control.matlab import *
+import matplotlib.pyplot as plt
+import collections
+import _collections_abc
+import os
+
+
+class DifferenceEquation:
+    def __init__(
+        self, tf_numerator_coefficients: list, tf_denominator_coefficients: list
+    ):
+
+        """ Creates a difference equation based on the input discrete domain transfer function
+
+        :list tf_numerator_coefficients: The coefficients of the numerator of the discrete transfer function
+        :list tf_denominator_coefficients: The coefficients of the denominator of the discrete transfer function
+        """
+        self.__numerator_coefficients = tf_numerator_coefficients
+        self.__denominator_coefficients = tf_denominator_coefficients
+
+        # Store the order of the numerator and denominator for calculation later
+        self.__controller_rank = len(self.__denominator_coefficients)
+        self.__input_coeffs_order = len(self.__numerator_coefficients)
+
+        # Normalize by the coefficient of the highest order denominator term
+        # This is done so that the output of the difference equation is not scaled
+        for i in range(0, len(self.__numerator_coefficients)):
+            self.__numerator_coefficients[i] /= self.__denominator_coefficients[0]
+
+        for i in range(0, len(self.__denominator_coefficients)):
+            self.__denominator_coefficients[i] /= self.__denominator_coefficients[0]
+
+        # We must keep track of the system state and previous states
+        self.time_step_count = 0
```

Comment:
Should `time_step_count` also be "private", ie. `__time_step_count`?

Reply:
Yep, made it 'private' and added a getter function 👍